packages/webpack-cli/lib/groups/HelpGroup.js

```diff
@@ -3,8 +3,16 @@ const { core, commands } = require('../utils/cli-flags');
 const commandLineUsage = require('command-line-usage');
 
 class HelpGroup {
-    outputHelp(isCommand = true, subject) {
-        if (subject) {
+    outputHelp(isCommand = true, subject, args) {
+        const [, , ...rawArgs] = args;
+        const commandsUsed = rawArgs.filter((val) => commands.find(({ name }) => name === val));
+        const flagsUsed = rawArgs.filter((val) => core.find(({ name }) => name === val.slice(2)));
+        const argsUsed = [...commandsUsed, ...flagsUsed];
+        const invalidArgs = rawArgs.filter(
+            (e) => !argsUsed.includes(e) && !e.includes('--color') && e !== 'version' && e !== '-v' && e !== 'help',
+        );
```

Comment:
To be honestly it is very dirty and potential buggy solution, we can parse them using own util https://github.com/webpack/webpack-cli/blob/3a05fcab490c028446c2cda83ef06fe3711647d1/packages/webpack-cli/lib/bootstrap.js#L60